src/kudu/rpc/client_negotiation.cc

```diff
@@ -230,6 +240,42 @@ Status ClientNegotiation::Negotiate(unique_ptr<ErrorStatusPB>* rpc_error) {
   return Status::OK();
 }
 
+Status ClientNegotiation::HandleTLS() {
+  if (encryption_ == RpcEncryption::DISABLED) {
+    return Status::NotSupported("RPC encryption is disabled.");
+  }
+
+  if (!tls_context_->has_signed_cert()) {
+    return Status::NotSupported("A signed certificate is not available.");
+  }
+
+  client_features_ = kSupportedClientRpcFeatureFlags;
+  client_features_.insert(TLS);
+  server_features_ = kSupportedServerRpcFeatureFlags;
+  server_features_.insert(TLS);
+  negotiated_authn_ = AuthenticationType::CERTIFICATE;
+
+  RETURN_NOT_OK(((security::TlsContext*)tls_context_)->SetSupportedAlpns(kAlpns, false));
```

Comment:
Can we add comment that this is Application layer protocol negotiation.
Also needs an explanation why this is required.

Reply:
Added comments in tls_context.h.